rviz_common/src/rviz_common/visualization_frame.cpp

```diff
@@ -738,16 +739,52 @@ void VisualizationFrame::setDisplayConfigModified()
   }
 }
 
+void VisualizationFrame::setDisplayTitleFormat(const QString & title_format)
+{
+  display_title_format_ = title_format.toStdString();
+}
+
 void VisualizationFrame::setDisplayConfigFile(const std::string & path)
 {
   display_config_file_ = path;
-
   std::string title;
-  if (path == default_display_config_file_) {
-    title = "RViz[*]";
+
+  if (display_title_format_.empty()) {
+    if (path == default_display_config_file_) {
+      title = "RViz[*]";
+    } else {
+      title = QDir::toNativeSeparators(QString::fromStdString(path)).toStdString() + "[*] - RViz";
+    }
+
+    /// Adding a namespace prefix on the window frame title
+    std::string node_namespace;
+    node_namespace = rviz_ros_node_.lock()->get_raw_node()->get_namespace();
+    if (node_namespace.compare("/") != 0) {
+      title = node_namespace + " - " + title;
+    }
```

Comment:
I don't think we should add this by default.  I think that the title is already too long.  Let's remove this for now, since the rest of the PR already adds in the ability for the user to set it to what they want.